leo/commands/setup.rs

```diff
@@ -64,75 +61,75 @@ impl Command for Setup {
-                        Groth16::<Bls12_377, Compiler<Fr, _>, Vec<Fr>>::setup(&program, rng)?;
-
-                    // TODO (howardwu): Convert parameters to a 'proving key' struct for serialization.
-                    // Write the proving key file to the output directory
-                    let proving_key_file = ProvingKeyFile::new(&package_name);
-                    tracing::info!("Saving proving key ({:?})", proving_key_file.full_path(&path));
-                    let mut proving_key_bytes = vec![];
-                    proving_key.write(&mut proving_key_bytes)?;
-                    let _ = proving_key_file.write_to(&path, &proving_key_bytes)?;
-                    tracing::info!("Complete");
-
-                    // Write the verification key file to the output directory
-                    let verification_key_file = VerificationKeyFile::new(&package_name);
-                    tracing::info!("Saving verification key ({:?})", verification_key_file.full_path(&path));
-                    let mut verification_key = vec![];
-                    proving_key.vk.write(&mut verification_key)?;
-                    let _ = verification_key_file.write_to(&path, &verification_key)?;
-                    tracing::info!("Complete");
-
-                    (proving_key, prepared_verifying_key)
-                } else {
-                    tracing::info!("Detected saved setup");
-
-                    // Read the proving key file from the output directory
-                    tracing::info!("Loading proving key...");
-
-                    if self.skip_key_check {
-                        tracing::info!("Skipping curve check");
-                    }
-                    let proving_key_bytes = ProvingKeyFile::new(&package_name).read_from(&path)?;
-                    let proving_key =
-                        Parameters::<Bls12_377>::read(proving_key_bytes.as_slice(), !self.skip_key_check)?;
-                    tracing::info!("Complete");
-
-                    // Read the verification key file from the output directory
-                    tracing::info!("Loading verification key...");
-                    let verifying_key_bytes = VerificationKeyFile::new(&package_name).read_from(&path)?;
-                    let verifying_key = VerifyingKey::<Bls12_377>::read(verifying_key_bytes.as_slice())?;
-
-                    // Derive the prepared verifying key file from the verifying key
-                    let prepared_verifying_key = PreparedVerifyingKey::<Bls12_377>::from(verifying_key);
-                    tracing::info!("Complete");
-
-                    (proving_key, prepared_verifying_key)
-                };
-
-                Ok((program, proving_key, prepared_verifying_key))
-            }
-            None => {
-                let mut main_file_path = path;
-                main_file_path.push(SOURCE_DIRECTORY_NAME);
-                main_file_path.push(MAIN_FILENAME);
+        if input.is_none() {
```

Comment:
Can you explain the change in this file? Not quite following whats going on.

Reply:
Oh. It's simple - instead of making huge match block with unnecessary indentation I check is_none and then proceed with non-none result.